components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/self-registration-process.jsp

```diff
@@ -138,6 +139,16 @@
             try {
 
                 for (Claim claim : claims) {
+
+                    if(claim.getUri().trim().contains("/claims/locality") &&
+                        StringUtils.isBlank(request.getParameter(claim.getUri()))){
```

Comment:
Apart from current conditions check whether "Accept-Language" header is not empty